android/java/org/chromium/chrome/browser/app/domain/NetworkModel.java

```diff
@@ -284,6 +284,20 @@ public void init() {
                 return;
             }
 
+            // Mark hidden networks as visible in preferences.
+            for (Map.Entry<String, Integer> entry :
+                    WalletConstants.KNOWN_TEST_CHAINS_MAP.entrySet()) {
+                if ((!AndroidUtils.isDebugBuild())
+                        && entry.getKey().equals(BraveWalletConstants.LOCALHOST_CHAIN_ID)) {
+                    // Hide local host for non-debug builds.
+                    mJsonRpcService.addHiddenNetwork(
+                            entry.getValue(), entry.getKey(), result -> {/* No-op. */});
+                } else {
+                    mJsonRpcService.removeHiddenNetwork(
+                            entry.getValue(), entry.getKey(), result -> {/* No-op. */});
```

Comment:
Why we need this now?

Reply:
Also `AndroidUtils.isDebugBuild()` is not going to change for particular browser installation, so this becomes some android-specific wallet setup. Can we do that in backend behind `#ifdef OS_ANDROID` in `GetDefaultHiddenNetworks` ?